### PR TITLE
FOUNDATIONS: Retrieve All Groups

### PR DESCRIPTION
**Pull request by evangelinedrink**

### CODE REVIEW

File: Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Groups/GroupServiceTests.Exceptions.RetrieveAll.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+            this.apiBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowDependencyExceptionOnRetrieveAllIfDependencyApiErrorOccursAndLogItAsync()
+        {
+            //given
+            var randomExceptionMessage = GetRandomMessage();
```
Comment (ShriHumrudha) [inline Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Groups/GroupServiceTests.Exceptions.RetrieveAll.cs:62]: string randomExceptionMessage

File: Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Groups/GroupServiceTests.Logic.RetrieveAll.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            retrievedGroups.Should().BeEquivalentTo(expectedGroups);
+
+            this.apiBrokerMock.Verify(broker =>
+                broker.GetAllGroupsAsync(),
+                    Times.Once());
+
+            this.apiBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+        }
+
```
Comment (ShriHumrudha) [inline Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Groups/GroupServiceTests.Logic.RetrieveAll.cs:44]: Remove these extra lines

File: Taarafo.Portal.Web/Models/Groups/Exceptions/GroupServiceException.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+// ---------------------------------------------------------------
+
+using System;
+using Xeptions;
+
+namespace Taarafo.Portal.Web.Models.Groups.Exceptions
+{
+    public class GroupServiceException : Xeption
+    {
+        public GroupServiceException(Exception innerException)
```
Comment (ShriHumrudha) [inline Taarafo.Portal.Web/Models/Groups/Exceptions/GroupServiceException.cs:13]: Xeption innerException

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Groups/GroupServiceTests.Exceptions.RetrieveAll.cs (hunk 1 above)
Comment (evangelinedrink) [reply Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Groups/GroupServiceTests.Exceptions.RetrieveAll.cs]: I changed the variable type to string.

File: Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Groups/GroupServiceTests.Logic.RetrieveAll.cs (hunk 2 above)
Comment (evangelinedrink) [reply Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Groups/GroupServiceTests.Logic.RetrieveAll.cs]: I removed the extra lines.

File: Taarafo.Portal.Web/Models/Groups/Exceptions/GroupServiceException.cs (hunk 3 above)
Comment (evangelinedrink) [reply Taarafo.Portal.Web/Models/Groups/Exceptions/GroupServiceException.cs]: I changed the Exception to Xeption in the PR titled "CODE RUB: Changed Exception to Xeption."

**Review by ShriHumrudha — APPROVED** [review]

**Pull request merged**